applications/StructuralMechanicsApplication/custom_processes/distribute_load_on_surface_process.cpp

```diff
@@ -48,18 +50,23 @@ void DistributeLoadOnSurfaceProcess::ExecuteInitializeSolutionStep()
 
     IntervalUtility interval_utility(mParameters);
     if (interval_utility.IsInInterval(current_time)) {
-        double total_area = 0.0;
-        for (auto& r_cond : mrModelPart.Conditions()) {
-            total_area += r_cond.GetGeometry().Area();
-        }
+        // Calculate the total area
+        const auto& r_communicator = mrModelPart.GetCommunicator();
+        const auto& r_local_mesh = r_communicator.LocalMesh();
+        const auto& r_conditions_array = r_local_mesh.Conditions();
```

Comment:
I think there is no need to take the local mesh as conditions will always be local.